chispa/number_helpers.py

```diff
@@ -1,4 +1,5 @@
 import math
+from typing import Optional
```

Comment:
Does this require a specific Python version?

Reply:
Hi @MrPowers , cheers for taking an initial look. 

I think so... it's happening [on this line](https://github.com/MrPowers/chispa/blob/05339042c843e1aef8dabbd2733ccd794201e606/chispa/number_helpers.py#L27). I don't think I changed the test for floating numbers which was already there, only moved it. I think the values coming into the `check_equal` function will be core Python types, so there won't be a difference between pyspark datatypes: `float`, `double` and `decimal`, they'll just all be interpreted as Python `'float'` and so there will be an approximate comparison. When it comes to `DecimalType` is it fine to keep the approximate comparison, or should an exact quality comparison be made (since it isn't subject to floating point errors)?

Reply:
In response to your 2nd comment. Yes, the `typing` module requires Python version 3.5. Does that present a compatibility problem? If you want I could remove it for now. Perhaps it could be a solution to maintain a backwards compatible branch for a while that still gets some of the new features, with the main branch refreshed with some additional type hinting. Might be more effort than it's worth though - let me know what you want to go with.